File: 2342-max-sum-of-a-pair-with-equal-sum-of-digits/2342-max-sum-of-a-pair-with-equal-sum-of-digits.go
func maximumSum(nums []int) int {
	arr := [82]int{}
	ans := -1
	var s, n int
	for _, num := range nums {
		s, n = 0, num
		for n > 0 {
			s += n % 10
			n /= 10
		}
		if arr[s] != 0 && ans < num+arr[s] {
			ans = num + arr[s]
		}
		if arr[s] < num {
			arr[s] = num
		}
	}
	return ans
}

File: 2342-max-sum-of-a-pair-with-equal-sum-of-digits/2342-max-sum-of-a-pair-with-equal-sum-of-digits.py
class Solution:
  def maximumSum(self, nums):
    m, ans = {}, -1
    for num in nums:
      s, n = 0, num
      while n:
        s += n % 10
        n //= 10
      if s not in m:
        m[s] = num
      else:
        ans = max(ans, num + m[s])
        m[s] = max(m[s], num)
    return ans
